rfd/XXXX-securing-amis.md

```diff
@@ -0,0 +1,225 @@
+---
+authors: Trent Clarke (trent@goteleport.com)
+state: Draft
+---
+
+# RFD 0XXX - Delivering more secure AMIs
+
+# Required Approvers
+* Engineering: @r0mant
+* Security: (@reedloden && @jentfoo)
+* Product: (@xinding33 || @klizhentas)
+
+## What
+
+This RFD discusses structures and processes to increase the security of the
+Amazon EC2 Disk Images (a.k.a AMIs) used to ship Teleport to customers.
+
+## Why
+
+One way our customers consume Teleport is by using a pre-built and  -configured
+AMI to run Teleport in AWS EC2 instances. As a provider of these images we are 
+(at least partially) responsible for the secutity of all ssoftware included in
+the image, not just teleport. 
+
+We should endeavour to not ship vulnerabilities to our clients, even if those
+vulnerabilities are not directly in our software.
+
+This is not simply an academic exercise. Customers have been asking for hardened
+AMIs - especially sor AMIs that conform to well-known security beckhmarks. See 
+[here](https://github.com/gravitational/teleport/issues/8078) for an example.
+
+## Details
+
+### Current Situation
+
+We  currently use [Packer](https://www.hashicorp.com/products/packer) and a shell
+script to provision our AMIs, based on Amazon Linux 2. The same Packerfile is used
+as a basis for several AMIs
+
+Inclusions for all AMIs:
+  * [Telegraf](https://www.influxdata.com/time-series-platform/telegraf/)
```

Comment:
Over to you, @webvictim. The commit date is 4 years ago, so I'm sure you remember it clearly :-)

Reply:
History.  We had the TIG stack baked into Gravity's `monitoring-app`. Though we eventually ripped it out and replaced it with Prometheus: https://github.com/gravitational/monitoring-app/pull/122.

I'm sure at the time these were put together, we chose to go along with what we were already comfortable with, and then those choices were never questioned until now.

Reply:
I really want to change this, but until we work on https://github.com/gravitational/teleport/issues/2734 and basically all the other issues that are holding Terraform back (see epic: https://github.com/gravitational/teleport/issues/24785) we're stuck with this stack. If we don't mind a breaking change to the monitoring setup for the few people who might actually use it, we could eliminate it.

My impression was that @hugoShaka was going to work on Terraform after the v12 Helm chart improvements, but unfortunately that hasn't panned out so far.

Reply:
See: https://github.com/gravitational/teleport/issues/2734

TIG stack is technical debt, IMO we should start an AWS managed prometheus instance instead of doing our own thing.